src_c/mixer.c

```diff
@@ -343,20 +345,42 @@ _init(int freq, int size, int stereo, int chunk, char *devicename)
     if (!size) {
         size = request_size;
     }
-    if (!stereo) {
-        stereo = request_stereo;
+
+    if (!channels) {
+        channels = request_channels;
+    }
+#if IS_SDLv1
+    switch (channels) {
+        case 1:
+        case 2:
+            break;
+        default:
+            PyErr_SetString(PyExc_ValueError, "'channels' must be 1 (mono) or 2 (stereo)");
+            return NULL;
     }
+#else /* IS_SDLv2 */
+    switch (channels) {
+        case 1:
+        case 2:
+        case 4:
+        case 6:
+            break;
+        default:
+            PyErr_SetString(PyExc_ValueError, "'channels' must be 1, 2, 4, or 6");
+            return NULL;
+    }
+#endif /* IS_SDLv2 */
+
     if (!chunk) {
         chunk = request_chunksize;
     }
 
     if (!devicename) {
         devicename = request_devicename;
     }
-    if (stereo >= 2)
-        stereo = 2;
-    else
-        stereo = 1;
+    if (allowedchanges == -1) {
+        allowedchanges = request_allowedchanges;
+    }
```

Comment:
Should there be a check/conversion from `PYGAME_MIXER_DEFAULT_ALLOWEDCHANGES` to `AUDIO_ALLOW_ANY_CHANGE`/`SDL_AUDIO_ALLOW_ANY_CHANGE` here?

Reply:
`PYGAME_MIXER_DEFAULT_ALLOWEDCHANGES` should be set to 0 or any combination of those flags, so it seems right

Reply:
You are right. For some reason I thought 0 wasn't an acceptable value because there was no SDL define for it, but it just means no changes allowed.